core/network/impl/synchronizer_impl.cpp

```diff
@@ -927,7 +942,7 @@ namespace kagome::network {
       }
       BlocksRequest request{
           .fields = BlockAttribute::HEADER | BlockAttribute::JUSTIFICATION,
-          .from = root.hash,
+          .from = known_blocks_.at(root.hash).data.header.value().parent_hash,
```

Comment:
`.at()` potential source of exception. Does it guaranteed `known_blocks_` contains `root.hash`? Use `op[]` if **yes**, or add check up otherwise.

Reply:
`.at()` was used intentionally, because `std::unordered_map::operator[]` doesn't assert.